scripts/quests/otherAreas/Knocking_on_Forbidden_Doors.lua

```diff
@@ -0,0 +1,277 @@
+                end,
+            },
+
+            onEventFinish =
+            {
+                [536] = function(player, csid, option, npc)
+                    quest:setVar(player, 'prog', 1)
+                end,
+            },
+        },
+
+        [xi.zone.PHOMIUNA_AQUEDUCTS] =
+        {
+            ['Wooden_Ladder'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'prog') == 1 and
+                    npc:getID() == ID.npc.LADDER_KNOCKING then -- Find a better way to identify exactly which ladder it is in case IDs shift
+                        return quest:progressEvent(38, 27) -- Essentially chains the ladder cs with the actual quest cs
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [38] = function(player, csid, option, npc)
+                    quest:setVar(player, 'prog', 2)
+                end,
+            },
+        },
+
+        [xi.zone.MISAREAUX_COAST] =
+        {
+            ['qm_mire_incense'] =
+            {
+                onTrigger = function(player, npc)
+                    --- ??? by end of river
+                    if quest:getVar(player, 'prog') == 2 and
+                    not player:hasKeyItem(xi.ki.MIRE_INCENSE) then
+                        player:addKeyItem(xi.ki.MIRE_INCENSE)
+                        return quest:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.MIRE_INCENSE)
+                    end
+                end,
+            },
+
+            ['qm_alsha'] =
+            {
+                onTrigger = function(player, npc)
+                    -- First time clicking on ???
+                    if player:hasKeyItem(xi.ki.MIRE_INCENSE) and
+                    quest:getVar(player, 'prog') == 2 then
+                        return quest:progressEvent(556, { [1] = xi.ki.MIRE_INCENSE })
+
+                    -- Second time clicking on ???
+                    elseif quest:getVar(player, 'prog') == 3 then
+                        return quest:progressEvent(557, { [1] = xi.ki.MIRE_INCENSE })
+
+                    -- Clicking on the ??? after killing NM
+                    elseif quest:getVar(player, 'prog') == 4 then
+                        return quest:event(558)
+                    end
+                end,
+            },
+
+            ['Alsha'] =
+            {
+                onMobDeath = function(mob, player, isKiller, firstCall)
+                    if quest:getVar(player, 'prog') == 3 and isKiller ~= mob then
+                        quest:setVar(player, 'prog', 4)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [556] = function (player, csid, option, npc)
+                    quest:setVar(player, 'prog', 3)
+                end,
+
+                [557] = function (player, csid, option, npc)
+                    if quest:getVar(player, 'prog') == 3 and
```

Comment:
Multiline condition format

Reply:
when you review this one next: I took the multiline off this one because I don't have any code in place for her to be able to cast fire on herself anyway, so no point yet in checking whether or not she nuked herself to 0.